compiler/Eta/Main/DriverPipeline.hs

```diff
@@ -219,6 +220,8 @@ compileOne' m_tc_result mHscMessage
                     hm_details = details,
                     hm_iface = iface,
                     hm_linkable = maybe_old_linkable }
+            when (hsc_lang == HscInterpreted) $
+              addModuleClassPath hsc_env (concatMap linkableObjs (maybeToList maybe_old_linkable))
```

Comment:
If I'm reading this correctly - `addModuleClassPath` will get called for every module that is being compiled when interepreted mode is going on. Perhaps a more efficient way to get it done is here: https://github.com/typelead/eta/blob/57ec8b168375c64abbec1fa65c15a3a912142cd5/compiler/Eta/REPL/Linker.hs#L309

Reply:
Mmm you mean using the function `linkClasses`? Sorry if if missing something obvious, how would improve the loading?

Reply:
Improving in the sense that you reduce the number of calls to `eta-serv`. If you do it inside of `unload_wkr`, which gets called once per build (regardless of the number of modules) and it will be a lot more efficient.

Reply:
Ok, so the idea is accumulate the non compiled modules and call iserv just one time with all the classes/jars of all, right?
Actually the `unload` action is called with `keep_linkables` empty, although the modules are not been recompiled. I get no way to trigger a call with some info in that param.
I've added some more traces to check it. 
So we should add some code in [`GHCMake`](https://github.com/typelead/eta/blob/57ec8b168375c64abbec1fa65c15a3a912142cd5/compiler/Eta/Main/GhcMake.hs#L309) to add those modules "up-to-date" to the call to `unload`, right?

Reply:
However, in the traces from build+repl i see:
```
Stable obj: [$37V :-> Primes, $37Z :-> Main]
Stable BCO: []
Unload old objects
Keeping linkables:  []
Adding module classpath: []
```
So in this code from GHCMake: 
```haskell
liftIO $ debugTraceMsg dflags 2 (text "Stable obj:" <+> ppr stable_obj $$
                            text "Stable BCO:" <+> ppr stable_bco)

    -- Unload any modules which are going to be re-linked this time around.
    let stable_linkables = [ linkable
                           | m <- nonDetEltsUniqSet stable_obj ++
                                  nonDetEltsUniqSet stable_bco,
                             -- It's OK to use nonDetEltsUniqSet here
                             -- because it only affects linking. Besides
                             -- this list only serves as a poor man's set.
                             Just hmi <- [lookupHpt pruned_hpt m],
                             Just linkable <- [hm_linkable hmi] ]
    liftIO $ unload hsc_env stable_linkables
```
`lookupHpt pruned_hpt` is returning []?

Reply:
@rahulmutt maybe we can left the optimization for later
I was thinking taht maybe we could load all stable modules?

Reply:
Yes, that's what I was thinking as well, but it looks like something's amiss in the stable module calculations since it's showing `Keeping linkables: []`.